pkg/driver/node/credentialprovider/provider_test.go

```diff
@@ -472,3 +473,199 @@ func TestCredentialFallbackEmptySecretData(t *testing.T) {
+		},
+
+		// Secret key with base64 characters
+		{
+			name:        "valid secret key with base64 characters",
+			accessKeyID: "AKIAIOSFODNN7EXAMPLE",
+			secretKey:   "wJalrXUtnFEMI/K7MDENG+bPxRfiCY/EXAMPLEKEY==",
+			expectError: false,
+			description: "Secret keys with base64 characters (/, +, =) should be accepted",
+		},
+
+		// Maximum length (128 characters)
+		{
+			name:        "valid 128-character access key (maximum allowed)",
+			accessKeyID: "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQR", // 128 chars
+			secretKey:   "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQR",
+			expectError: false,
+			description: "128-character keys should be accepted (maximum limit)",
+		},
+
+		// Invalid: exceeds maximum length
+		{
+			name:        "invalid access key exceeds 128 characters",
+			accessKeyID: "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTU", // 129 chars
+			secretKey:   "TESTSECRET456",
+			expectError: true,
+			description: "Access keys exceeding 128 characters should be rejected",
+		},
+		{
+			name:        "invalid secret key exceeds 128 characters",
+			accessKeyID: "AKIAIOSFODNN7EXAMPLE",
+			secretKey:   "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTU", // 129 chars
+			expectError: true,
+			description: "Secret keys exceeding 128 characters should be rejected",
+		},
+
+		// Invalid: special characters (not alphanumeric or base64)
+		{
+			name:        "invalid access key with special characters",
+			accessKeyID: "INVALID@KEY!",
+			secretKey:   "TESTSECRET456",
+			expectError: true,
+			description: "Access keys with special characters should be rejected",
+		},
+		{
+			name:        "invalid access key with spaces",
+			accessKeyID: "INVALID KEY",
+			secretKey:   "TESTSECRET456",
+			expectError: true,
+			description: "Access keys with spaces should be rejected",
+		},
+		{
+			name:        "invalid secret key with invalid special characters",
+			accessKeyID: "AKIAIOSFODNN7EXAMPLE",
+			secretKey:   "INVALID@SECRET!",
+			expectError: true,
+			description: "Secret keys with invalid special characters should be rejected",
+		},
+
+		// Edge cases: empty strings
+		{
+			name:        "invalid empty access key",
+			accessKeyID: "",
+			secretKey:   "TESTSECRET456",
+			expectError: true,
+			description: "Empty access key should be rejected",
+		},
+		{
+			name:        "invalid empty secret key",
+			accessKeyID: "AKIAIOSFODNN7EXAMPLE",
+			secretKey:   "",
+			expectError: true,
+			description: "Empty secret key should be rejected",
+		},
+
+		// Edge cases: whitespace handling
+		{
+			name:        "valid access key with leading/trailing spaces (trimmed)",
```

Comment:
Feeling nitpicky: you could also add a test with the maximum length allowed AND extra leading/trailing spaces, to check that the length is checked after trimming.